src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Workspace.cs

```diff
@@ -0,0 +1,510 @@
+
+            snapshot.Properties.TryGetValue(ConfigurationGeneral.LanguageServiceNameProperty, out string? languageName);
+            snapshot.Properties.TryGetValue(ConfigurationGeneral.TargetPathProperty, out string? binOutputPath);
+            snapshot.Properties.TryGetValue(ConfigurationGeneral.MSBuildProjectFullPathProperty, out string? projectFilePath);
+            snapshot.Properties.TryGetValue(ConfigurationGeneral.AssemblyNameProperty, out string? assemblyName);
+            snapshot.Properties.TryGetValue(ConfigurationGeneral.CommandLineArgsForDesignTimeEvaluationProperty, out string? commandLineArgsForDesignTimeEvaluation);
+
+            if (string.IsNullOrEmpty(languageName) || string.IsNullOrEmpty(binOutputPath) || string.IsNullOrEmpty(projectFilePath))
+            {
+                // Insufficient data to initialize the language service.
+                Interlocked.Exchange(ref _state, StateFailed);
+
+                Exception ex = new("Insufficient project data to initialize the language service.");
+                _initializationCompletion.TrySetException(ex);
+
+                throw ex;
+            }
+
+            try
+            {
+                // Note that we create operation progress registrations using the first primary (active) configuration
+                // within the slice. Over time this may change, but we keep the same registration to the first seen.
+                ConfiguredProject configuredProject = evaluationUpdate.Value.ConfiguredProject;
+
+                _evaluationProgressRegistration = _dataProgressTrackerService.RegisterForIntelliSense(this, configuredProject, "LanguageServiceHost.Workspace.Evaluation");
+                _buildProgressRegistration = _dataProgressTrackerService.RegisterForIntelliSense(this, configuredProject, "LanguageServiceHost.Workspace.ProjectBuild");
+
+                _disposableBag.Add(_evaluationProgressRegistration);
+                _disposableBag.Add(_buildProgressRegistration);
+
+                _contextId = GetWorkspaceProjectContextId(projectFilePath, _projectGuid, _slice);
+
+                _disposableBag.Add(_activeEditorContextTracker.RegisterContext(_contextId));
+
+                object? hostObject = _unconfiguredProject.Services.HostObject;
+
+                // Call into Roslyn to init language service for this project
+                _context = await _workspaceProjectContextFactory.Value.CreateProjectContextAsync(
+                    languageName,
+                    _contextId,
+                    projectFilePath,
+                    _projectGuid,
+                    hostObject,
+                    binOutputPath,
+                    assemblyName,
+                    cancellationToken);
+
+                _disposableBag.Add(_context);
+
+                // Update additional properties within a batch to avoid thread pool starvation.
+                // https://github.com/dotnet/project-system/issues/8027
```

Comment:
Looks like this issue was closed.

Reply:
Yeah. This comment was added in the PR that closed it: https://github.com/dotnet/project-system/pull/8031/files#diff-1844e8887b42f22e08839aadff0e0e290a54b9e62bfc316f7c3fdb851d502cbcR82-R83